### PR TITLE
Functional tests - Sort suppliers table

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Sort of suppliers table
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  |  TEST_PATH="functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19004)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    await testContext.addContextItem(this, 'testIdentifier', 'goToSuppliersPage', baseContext);
+    await this.pageObjects.brandsPage.goToSubTabSuppliers();
+    const pageTitle = await this.pageObjects.suppliersPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.suppliersPage.pageTitle);
+  });
+  // 1: Create two suppliers
+  describe('Create 2 suppliers', async () => {
+    const tests = [
+      {args: {supplierToCreate: firstSupplierData}},
+      {args: {supplierToCreate: secondSupplierData}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js:82]: You're using only 2 suppliers ? Is it enough to verify ?

File: tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    ];
+    tests.forEach((test, index) => {
+      it('should filter supplier by name', async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `filterToDelete${index + 1}`, baseContext);
+        await this.pageObjects.suppliersPage.filterTable(
+          'input',
+          'name',
+          test.args.supplierData.name,
+        );
+        // Check number od suppliers
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js]:
```suggestion
        // Check number of suppliers
```

File: tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js:82]: I can add more suppliers, but I think if the sort of 2 different suppliers is passed correctly also the sort of 3,4,5... will be sorted correctly.

Comment (SimonGrn) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js:82]: Could you add a third one please ? That way we're sure that a sort is correct. It's impossible to verify with only 2.

Comment (nesrineabdmouleh) [reply tests/puppeteer/campaigns/functional/BO/03_catalog/05_brandsAndSuppliers/suppliers/04_sortSuppliers.js:82]: Done :)

**Review by SimonGrn — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**